prov/verbs/src/verbs_ep.c

```diff
@@ -240,10 +240,16 @@ static int fi_ibv_ep_close(fid_t fid)
 
 	switch (ep->util_ep.type) {
 	case FI_EP_MSG:
+		if (ep->eq)
+			fastlock_acquire(&ep->eq->lock);
+
 		if (fi_ibv_is_xrc(ep->info))
 			fi_ibv_ep_xrc_close(ep);
 		else
 			rdma_destroy_ep(ep->id);
+
```

Comment:
This will cause a deadlock with XRC when a message endpoint is closed (if it was not ever fully connected, like a simultaneous RXM reject).

To avoid the deadlock for XRC codepath it will make more sense to take the lock inside of fi_ibv_ep_xrc_close() right before the call to fi_ibv_ep_destroy_xrc_qp(), and release it before fi_ibv_ep_xrc_close() returns.

Reply:
@a-ilango - Once the changes I have in PR #5016 are accepted, this lock can be taken where you originally have it (around the if/else clause). I included the locking you have here in PR #5016 and made the necessary changes since the XRC EP close needed to be protected in my changes as well.

Reply:
Sounds good. I'll take a look at #5016 and let's merge that PR before this one.